src/tools/r2rdump/R2RReader.cs

```diff
@@ -113,33 +112,66 @@ public unsafe R2RReader(string filename)
                     R2RMethods = new List<R2RMethod>();
                     for (uint rid = 1; rid <= nMethodEntryPoints; rid++)
                     {
-                        uint offset = 0;
+                        int offset = 0;
                         if (methodEntryPoints.TryGetAt(_image, rid - 1, ref offset))
                         {
-                            R2RMethod method = new R2RMethod(_image, mdReader, methodEntryPoints, offset, rid);
-                            if (method.EntryPointRuntimeFunctionId >= nRuntimeFunctions)
+                            R2RMethod method = new R2RMethod(_image, mdReader, methodEntryPoints, rid, offset);
+
+                            if (method.EntryPointRuntimeFunctionId >= 0 && method.EntryPointRuntimeFunctionId < nRuntimeFunctions)
```

Comment:
How come you removed the throwing of the BadImageFormatException? Did you find some runtime function Id's that were greater than the number of runtime functions even though you had a valid image?